runtime/tools/python/ttrt/common/golden.py

```diff
@@ -148,11 +148,61 @@ def golden(binary, programContext, opContext):
             op_output_tensor, dtype=torch.float32
         ).flatten()
 
+        torch.save(golden_tensor_torch, f"{runtimeConfig.artifact_dir}/{loc}_golden.pt")
+        torch.save(output_tensor_torch, f"{runtimeConfig.artifact_dir}/{loc}_device.pt")
+
         _, _, cal_pcc, output_str = get_atol_rtol_pcc(
             golden_tensor_torch, output_tensor_torch
         )
 
         print(f"PCC={cal_pcc}")
         print(output_str)
+
+        results = {}
+        results["expected_pcc"] = runtimeConfig.pcc
+        results["actual_pcc"] = cal_pcc
+        results["atol"] = runtimeConfig.atol
+        results["rtol"] = runtimeConfig.rtol
+        results["allclose"] = torch.allclose(
+            golden_tensor_torch,
+            output_tensor_torch,
+            atol=runtimeConfig.atol,
+            rtol=runtimeConfig.rtol,
+        )
+        results["max"] = torch.max(
+            torch.abs(golden_tensor_torch - output_tensor_torch)
+        ).item()
+        results["mean_absolute_error"] = torch.mean(
+            torch.abs(golden_tensor_torch - output_tensor_torch)
+        ).item()
+        results["root_mean_square_error"] = torch.sqrt(
+            torch.mean((golden_tensor_torch - output_tensor_torch) ** 2)
+        ).item()
+        results["cosine_similarity"] = torch.nn.functional.cosine_similarity(
+            golden_tensor_torch.unsqueeze(0), output_tensor_torch.unsqueeze(0)
+        ).item()
+
+        # Create golden result json file if doesn't exist
```

Comment:
I think the file related stuff from this comment and below might be better to be refactored into a utility function, just so this golden function doesn't get too unwieldy.

Reply:
resolved and moved outside of this golden function